api/src/main/java/org/openmrs/module/xdssender/api/service/impl/XdsExportServiceImpl.java

```diff
@@ -2,89 +2,70 @@
 
 import org.dcm4chee.xds2.infoset.ihe.ProvideAndRegisterDocumentSetRequestType;
 import org.dcm4chee.xds2.infoset.rim.RegistryResponseType;
-import org.marc.everest.datatypes.II;
-import org.marc.everest.rmim.uv.cdar2.pocd_mt000040uv.Author;
 import org.openmrs.Encounter;
 import org.openmrs.Patient;
-import org.openmrs.Provider;
-import org.openmrs.api.context.Context;
 import org.openmrs.api.impl.BaseOpenmrsService;
 import org.openmrs.module.xdssender.XdsSenderConfig;
 import org.openmrs.module.xdssender.api.cda.ClinicalDocumentBuilder;
+import org.openmrs.module.xdssender.api.cda.ORM_O01DocumentBuilder;
 import org.openmrs.module.xdssender.api.cda.model.DocumentModel;
+import org.openmrs.module.xdssender.api.model.DocumentData;
 import org.openmrs.module.xdssender.api.model.DocumentInfo;
 import org.openmrs.module.xdssender.api.service.XdsExportService;
 import org.openmrs.module.xdssender.api.xds.MessageUtil;
 import org.openmrs.module.xdssender.api.xds.XdsSender;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.List;
-import java.util.UUID;
-
 @Component("xdsSender.XdsExportService")
 public class XdsExportServiceImpl extends BaseOpenmrsService implements XdsExportService {
-	
+
+	@Autowired
+	private ClinicalDocumentBuilder clinicalDocBuilder;
+
 	@Autowired
-	private ClinicalDocumentBuilder docBuilder;
-	
+	private ORM_O01DocumentBuilder ormDocBuilder;
+
 	@Autowired
 	private MessageUtil messageUtil;
-	
+
 	@Autowired
 	private XdsSenderConfig config;
-	
+
 	@Autowired
 	private XdsSender xdsSender;
-	
+
 	@Override
 	public DocumentInfo exportProvideAndRegister(Encounter encounter, Patient patient) {
 		try {
-			DocumentModel docModel = docBuilder.buildDocument(patient, encounter);
-			
-			DocumentInfo docInfo = buildDocInfo(encounter, patient, docModel);
-			
-			if (!messageUtil.getPatientIdentifier(docInfo).getIdentifierType().getName().equals("ECID")) {
+			DocumentModel clinicalDocModel = clinicalDocBuilder.buildDocument(patient, encounter);
+			DocumentInfo clinicalDocInfo = new DocumentInfo(encounter, patient, clinicalDocModel,
+					"text/xsl", config.getProviderRoot());
+			DocumentData clinicalDoc = new DocumentData(clinicalDocInfo, clinicalDocModel.getData());
+
+			DocumentData msgDoc = null;
+			DocumentModel msgDocModel = ormDocBuilder.buildDocument(encounter);
```

Comment:
This should only happen for orders that are being sent to SCC, see the logic in lab module (it's a concept check)

Reply:
I check this on api/src/main/java/org/openmrs/module/xdssender/api/cda/ORM_O01DocumentBuilder.java line 42.
Should this logic be moved or can it stay that way?

Reply:
ok that's fine - can you rename msgDoc and msgDocModel to labOrderDoc and labOrderDocModel?

Reply:
Ok, done.